packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/hooks.tsx

```diff
@@ -79,3 +79,25 @@ export const useDraftRequisitionLine = (item: ItemWithStatsFragment | null) => {
 
   return { draft, isLoading, save: () => draft && save(draft), update };
 };
+
+export const useNextRequestLine = (
+  currentItem: ItemWithStatsFragment | null
+) => {
+  const { lines } = useRequestRequisitionLines();
+
+  const nextState: {
+    isDisabled: boolean;
+    next: null | ItemWithStatsFragment;
+  } = { isDisabled: false, next: null };
+
+  const idx = lines.findIndex(l => l.item.id === currentItem?.id);
+  const next = lines[idx + 1];
+  if (!next) {
+    nextState.isDisabled = true;
+    return nextState;
+  }
+
+  nextState.next = next.item;
+
+  return nextState;
```

Comment:
Yeah the duplication pains me. I'm not seeing an easy way to make this generic as the use cases seem to be slightly different, though, this is likely going to be used a lot so maybe it's worth the time to try and do so

Reply:
heh. yeah - very similar but tricky to extract